### DISCUSSION TITLE
2021/01/16/midi-pulseaudio-linux/

### DISCUSSION THREAD

**Discussion by jdeluyck** [jdeluyck/kcoreorg-website#134] [Blog Comments]
# 2021/01/16/midi-pulseaudio-linux/
Learning the piano, and playing with MIDI & PulseAudio on Linux
https://kcore.org/2021/01/16/midi-pulseaudio-linux/

**Comment by jdeluyck** [comment]
Originally posted by Wayne Bonner on 2022-05-09T20:41:41Z

----

Bravo and thank you for this article.  
I bought a mini keyboard recently and was hoping for a nice plug\&play experience with Ubuntu 22.04.  
Instead I fell into the vortex of the PulseAudio/Alsa/Jack/Pipeline articles and wasted days trying to understand the systems, each with it's own commands and terminology.

All I wanted to do was make sounds with my keyboard!  
Your elegant simple solution solved my problem.  
Thanks

**Reply by jdeluyck** [reply]
You're welcome ;)

**Comment by denilsonsa** [comment]
I imagine this setup would be simpler now, using PipeWire instead of PulseAudio. [qpwgraph](https://gitlab.freedesktop.org/rncbc/qpwgraph) can be used to connect nodes in PipeWire, using a nice and intuitive GUI. I even wrote about how [PipeWire is awesome](https://denilson.sa.nom.br/blog/2023-11-06/pipewire-is-awesome) on my own personal blog.

**Reply by jdeluyck** [reply]
I haven't needed this setup for a while now - if I'd redo it it'll definitely be with PipeWire. Thanks for the link!
